packages/block-library/src/navigation/index.php

```diff
@@ -221,12 +221,38 @@ function( $block ) {
 		$is_fallback                      = true; // indicate we are rendering the fallback.
 		$attributes['__unstableMaxPages'] = 4; // set value to be passed as context to Page List block.
 
-		$page_list_block = array(
-			'blockName' => 'core/page-list',
-			'attrs'     => array(),
+		$navigation_posts = get_posts(
+			array(
+				'post_type' => 'wp_navigation',
+			)
 		);
 
-		$inner_blocks = new WP_Block_List( array( $page_list_block ), $attributes );
+		$navigation_post = null;
+
+		// Pick first non-empty Navigation.
+		foreach ( $navigation_posts as $navigation_maybe ) {
+			if ( ! empty( $navigation_maybe->post_content ) ) {
+				$navigation_post = $navigation_maybe;
+				break;
+			}
+		}
+
+		// Use non-empty Navigation if available.
+		if ( $navigation_post ) {
+			$fallback_blocks = parse_blocks( $navigation_post->post_content );
```

Comment:
What happens here if we don't have any blocks? 🤔

Reply:
Isn't that handled by the `! empty` in the loop above? Alternatively you could `parse_blocks` in the loop and check how many there are before selecting the post as a candidate.

Also, just be aware of the code on line 205-212 that filters the parsed blocks, that will probably have to be applied here too.

Reply:
> Alternatively you could parse_blocks in the loop and check how many there are before selecting the post as a candidate.

I was wondering whether parsing would be an expensive operation.

Reply:
Ok I avoided parsing but added a `null` check on `blockName`.

Reply:
I don't think it'd be too expensive. 

If the navigation post is empty, the parser won't have anything to do and should skip it. If the navigation post has content, the parser will have something to do, but that will very likely be the chosen post, so the parsing was necessary.

But having said that, it should generally be ok as it is. I can't think of a situation where the content might have only line breaks.